01_fib/willshen8/fib.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func calculateFirstNFibNumArray(n int) []uint64 {
+	fibResults := []uint64{1, 1}
+	for i := 2; i < n; i++ {
+		fibResults = append(fibResults, fibResults[i-1]+fibResults[i-2])
```

Comment:
`make` the result array with size `n` and current value with `fibResults[i]` - more run time efficient.

Reply:
Done!